cmd/client.go

```diff
@@ -25,106 +24,134 @@ import (
+	rootCmd.AddCommand(newCliCommand())
 }
 
 var (
 	opts       []string
-	pdAddress  string
 	startTs    uint64
 	targetTs   uint64
 	sinkURI    string
 	configFile string
+	cliPdAddr  string
+
+	cdcEtcdCli kv.CDCEtcdClient
+	pdCli      pd.Client
 )
 
-var cliCmd = &cobra.Command{
-	Use:   "cli",
-	Short: "simulate client to create changefeed",
-	Long:  ``,
-	RunE: func(cmd *cobra.Command, args []string) error {
-		ctx := context.Background()
-		etcdCli, err := clientv3.New(clientv3.Config{
-			Endpoints:   []string{pdAddress},
-			DialTimeout: 5 * time.Second,
-			DialOptions: []grpc.DialOption{
-				grpc.WithConnectParams(grpc.ConnectParams{
-					Backoff: backoff.Config{
-						BaseDelay:  time.Second,
-						Multiplier: 1.1,
-						Jitter:     0.1,
-						MaxDelay:   3 * time.Second,
-					},
-					MinConnectTimeout: 3 * time.Second,
-				}),
-			},
-		})
-		if err != nil {
-			return err
-		}
-		cli := kv.NewCDCEtcdClient(etcdCli)
-		pdCli, err := pd.NewClient([]string{pdAddress}, pd.SecurityOption{})
-		if err != nil {
-			return err
-		}
-		id := uuid.New().String()
-		if startTs == 0 {
-			ts, logical, err := pdCli.GetTS(ctx)
+func newCliCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "cli",
+		Short: "Manage replication task and TiCDC cluster",
+		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+			etcdCli, err := clientv3.New(clientv3.Config{
+				Endpoints:   []string{cliPdAddr},
+				DialTimeout: 5 * time.Second,
+				DialOptions: []grpc.DialOption{
+					grpc.WithConnectParams(grpc.ConnectParams{
+						Backoff: backoff.Config{
+							BaseDelay:  time.Second,
+							Multiplier: 1.1,
+							Jitter:     0.1,
+							MaxDelay:   3 * time.Second,
+						},
+						MinConnectTimeout: 3 * time.Second,
+					}),
+				},
+			})
 			if err != nil {
 				return err
 			}
-			startTs = oracle.ComposeTS(ts, logical)
-		}
-		err = verifyStartTs(ctx, startTs, cli)
-		if err != nil {
-			return err
-		}
+			cdcEtcdCli = kv.NewCDCEtcdClient(etcdCli)
+			pdCli, err = pd.NewClient([]string{cliPdAddr}, pd.SecurityOption{})
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+	command.AddCommand(
+		newCreateChangefeedCommand(),
+		newListCaptureCommand(),
+		newListChangefeedCommand(),
+		newListProcessorCommand(),
+		newQueryChangefeedCommand(),
+		newQueryProcessorCommand(),
+		newGetTsoCommand(),
+		newTruncateCommand(),
+	)
+	command.PersistentFlags().StringVar(&cliPdAddr, "pd", "http://127.0.0.1:2379", "PD address")
+
+	return command
+}
 
-		cfg := new(util.ReplicaConfig)
-		if len(configFile) > 0 {
-			if err := strictDecodeFile(configFile, "cdc", cfg); err != nil {
+func newCreateChangefeedCommand() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "create",
+		Short: "create a new replication task (changefeed)",
+		Long:  ``,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := context.Background()
+			id := uuid.New().String()
```

Comment:
maybe we can receive a change feed id.
if the user doesn't  specify a change feed id, we create a UUID as the default value

Reply:
note that we should check  the uniqueness of changefeedID if we allow the user to specify a changefeedID

Reply:
Why do you think a customized change feed ID is necessary? If it is necessary, maybe we can let the user give it a name and description?

Reply:
This PR mainly changes the interaction of the command line, I think we can refine the logic later.